### PR TITLE
Forge addKey() unique

### PR DESCRIPTION
**Pull request by natanfelles**
Allow to create UNIQUE KEYs with `addKey($key, $primary, $unique)`. Useful for multi-fields, arrays.

```php
$forge->addKey(['blog_id', 'uri'], false, true);
```

### CODE REVIEW

File: system/Database/Forge.php [hunk 1]
```diff
@@ -271,12 +277,13 @@ public function dropDatabase($db_name)
...
-	 * @param    string $key
-	 * @param    bool   $primary
+	 * @param    string|array $key
+	 * @param    bool         $primary
+	 * @param    bool         $unique
 	 *
 	 * @return    Forge
 	 */
-	public function addKey($key, $primary = false)
+	public function addKey($key, $primary = false, $unique = false)
```
Comment (samsonasik) [inline system/Database/Forge.php]: `$unique` can use `bool` type hint I think .: `bool $unique = false`

Comment (natanfelles) [reply system/Database/Forge.php]: Yes, and the `$primary` too because it is compared with `===`.

**Comment by lonnieezell** [comment]
I'm not a fan of adding another boolean here to further muddy the purpose of this method. Instead, implement a new method for adding unique keys. This should make it easier to read and know what's going on, and keep the code itself a bit cleaner.

**Comment by natanfelles** [comment]
`addUniqueKey($key)`?

And will make sense to add an `addPrimaryKey($key)` or it would be redundant?

**Comment by lonnieezell** [comment]
> `addUniqueKey($key)`?

Yes

> And will make sense to add an `addPrimaryKey($key)` or it would be redundant?

Actually, I'm all for adding this method.

**Comment by lonnieezell** [comment]
@natanfelles You forgot to update the docs for this.

**Comment by natanfelles** [comment]
Updated. But I do not found a more elegant way to work without adding the third param on `addKey()` yet.

**Comment by natanfelles** [comment]
\* more _elegant_ in sense to do not repeat the code in the Forge.

**Comment by lonnieezell** [comment]
@natanfelles What you did looks fine. Thank you.

**Pull request merged**